arcgis-ios-sdk-samples/Edit data/Edit features with feature-linked annotation/EditFeaturesWithFeatureLinkedAnnotationViewController.swift

```diff
@@ -0,0 +1,228 @@
+                                    // Show a message to select straight (single segment) polylines only.
+                                    self.presentAlert(title: "Make a different selection", message: "Select straight (single segment) polylines only.")
+                                    return
+                                }
+                            }
+                        }
+                        // Ensure the feature is not nil and select it.
+                        guard let selectedFeature = self.selectedFeature else { return }
+                        featureLayer.select(selectedFeature)
+                        // If the selected feature is a point, prompt the edit attributes alert.
+                        if selectedFeature.geometry?.geometryType.rawValue == 1 {
+                            self.showEditableAttributes(selectedFeature: selectedFeature)
+                        } else if selectedFeature.geometry?.geometryType.rawValue == 3 {
+                            // If the selected feature is a polyline, set the value to true.
+                            self.selectedFeatureIsPolyline = true
+                        }
+                    }
+                }
+            }
+        }
+    }
+    
+    // Create an alert dialog with edit texts to allow editing of the given feature's 'AD_ADDRESS' and 'ST_STR_NAM' attributes.
+    func showEditableAttributes(selectedFeature: AGSFeature) {
+        guard let selectedFeature = self.selectedFeature else { return }
+        // Create an alert controller and customize the title and message.
+        let alert = UIAlertController(title: "Edit feature attributes", message: "Edit the 'AD_ADDRESS' and 'ST_STR_NAM' attributes.", preferredStyle: .alert)
+        // Add text fields to prompt user input.
+        alert.addTextField()
+        alert.addTextField()
+        // Popuplate the text fields with the current values.
+        alert.textFields?[0].text = (selectedFeature.attributes["AD_ADDRESS"] as! NSNumber).stringValue
+        alert.textFields?[1].text = selectedFeature.attributes["ST_STR_NAM"] as? String
+        // Prompt the appropriate keyboard types.
+        alert.textFields?[0].keyboardType = .asciiCapableNumberPad
+        alert.textFields?[1].keyboardType = .asciiCapable
+        // Add a "Cancel" option and clear the selection if cancel is selected.
+        alert.addAction(UIAlertAction(title: "Cancel", style: .cancel) { _ in
+            self.clearSelection()
+            self.selectedFeature = nil
+        })
+        // Add a "Done" option to complete the editing process and close the alert.
+        alert.addAction(UIAlertAction(title: "Done", style: .default) { _ in
+            let addressTextField = alert.textFields?[0]
+            let streetTextField = alert.textFields?[1]
+            // Set the new values for the attributes and update the selected feature.
+            selectedFeature.attributes["AD_ADDRESS"] = Int((addressTextField?.text)!)
```

Comment:
The value held in `AGSFeature.attributes` cannot be an Int, only NSNumber -- so it was the best way to display the string value without it being an optional. And it is later casted as an `Int` because there was no `NSNumber`  initializer that would accept the value from the text field.

Reply:
I see. The latter case String to Int can utilize `NumberFormatter`, but it might be an overkill. Using Int.init should be fine. 👍